apps/hash-frontend/src/pages/types/types-page-tabs.tsx

```diff
@@ -0,0 +1,83 @@
+import { Tabs, tabsClasses, typographyClasses } from "@mui/material";
+import { FunctionComponent } from "react";
+
+import { TabLink } from "../../shared/ui/tab-link";
+import { TabId } from "./[[...type-kind]].page";
+
+const tabIds = [
+  "all",
+  "entity-type",
+  "link-type",
+  "property-type",
+  "data-type",
+] satisfies TabId[];
+
+export const tabTitles: Record<TabId, string> = {
+  all: "All",
+  "entity-type": "Entity Types",
+  "link-type": "Link Types",
+  "property-type": "Property Types",
+  "data-type": "Data Types",
+};
+
+type TypesPageTabsProps = {
+  currentTab: TabId;
+  numberOfTypesByTab: Record<TabId, number | undefined>;
+};
+
+export const TypesPageTabs: FunctionComponent<TypesPageTabsProps> = ({
+  currentTab,
+  numberOfTypesByTab,
+}) => {
+  return (
+    <Tabs
+      value={currentTab}
+      TabIndicatorProps={{
+        sx: ({ palette }) => ({
+          height: 3,
+          backgroundColor: palette.blue[60],
+          minHeight: 0,
+          bottom: -1,
+        }),
+      }}
+      sx={{
+        minHeight: 0,
+        overflow: "visible",
+        alignItems: "flex-end",
+        flex: 1,
+        [`.${tabsClasses.scroller}`]: {
+          overflow: "visible !important",
+        },
+      }}
+    >
+      {tabIds.map((tabId) => (
+        <TabLink
+          key={tabId}
+          value={tabId}
+          href={tabId === "all" ? "/types" : `/types/${tabId}`}
+          label={tabTitles[tabId]}
+          count={numberOfTypesByTab[tabId]}
+          loading={numberOfTypesByTab[tabId] === undefined}
```

Comment:
This doesn't work to show the loading indicator because there's a check for `typeof count === "number"` in `TabLink` before showing the loading indicator. 

It looks like we could remove that check and the `?? 0` from `count={entities?.length ?? 0}` on the entity type entities tab to get the desired behaviour (might be wrong)

Reply:
Ah thanks for catching, thought the easiest fix would to just add the `loading` boolean to the conditional. Have done so in https://github.com/hashintel/hash/pull/2908/commits/dd6059a9bc7ce8833451b21c3ab68c872611855f